components/brave_federated/data_stores/data_store.cc

```diff
@@ -73,9 +91,68 @@ bool DataStore::Init(int task_id,
   return db_.Open(database_path_) && EnsureTable();
 }
 
-DataStore::~DataStore() {}
+bool DataStore::AddTrainingInstance(
+    const mojom::TrainingInstancePtr training_instance) {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
 
-bool DataStore::DeleteLogs() {
+  sql::Statement s1(db_.GetUniqueStatement(
+      base::StringPrintf("SELECT MAX(training_instance_id) FROM %s",
+                         task_name_.c_str())
+          .c_str()));
+  s1.Step();
+  int training_instance_id = s1.ColumnInt(0) + 1;
+
+  base::Time created_at = base::Time::Now();
+
+  for (const auto& covariate : training_instance->covariates) {
+    sql::Statement s(db_.GetUniqueStatement(
+        base::StringPrintf(
+            "INSERT INTO %s (training_instance_id, feature_name, feature_type, "
+            "feature_value, created_at) "
+            "VALUES (?,?,?,?,?)",
+            task_name_.c_str())
+            .c_str()));
+
+    BindCovariateToStatement(*covariate, training_instance_id, created_at, &s);
+    s.Run();
+  }
+
+  return true;
+}
+
+DataStore::TrainingData DataStore::LoadTrainingData() {
+  DCHECK_CALLED_ON_VALID_SEQUENCE(sequence_checker_);
+
+  DataStore::TrainingData training_instances;
+  sql::Statement s(db_.GetUniqueStatement(
+      base::StringPrintf("SELECT id, training_instance_id, feature_name, "
+                         "feature_type, feature_value, "
+                         "created_at FROM %s",
+                         task_name_.c_str())
+          .c_str()));
+
+  training_instances.clear();
+  while (s.Step()) {
+    int training_instance_id = s.ColumnInt(1);
+    mojom::CovariatePtr covariate = mojom::Covariate::New();
+    covariate->covariate_type = (mojom::CovariateType)s.ColumnInt(2);
+    covariate->data_type = (mojom::DataType)s.ColumnInt(3);
+    covariate->value = s.ColumnString(4);
+
+    if (!base::Contains(training_instances, training_instance_id)) {
```

Comment:
This if statement is unnecessary.  We could call `training_instances[training_instance_id].push_back(*covariate);` without checking, because `training_instances[training_instance_id]` construction creates empty vector if it wasn't exist.